core/chains/evm/gas/block_history_estimator.go

```diff
@@ -128,11 +127,14 @@ func NewBlockHistoryEstimator(lggr logger.Logger, ethClient evmclient.Client, cf
 
 // OnNewLongestChain recalculates and sets global gas price if a sampled new head comes
 // in and we are not currently fetching
-func (b *BlockHistoryEstimator) OnNewLongestChain(_ context.Context, head *evmtypes.Head) {
+func (b *BlockHistoryEstimator) OnNewLongestChain(_ context.Context, head txmgrtypes.Head) {
+	// Get the native head object, using explicit cast
+	evmHead := head.(*evmtypes.Head)
```

Comment:
We need to avoid this cast one way one another. I don't think it's a problem to introduce a generic type param `[H Head]` for this purpose, but we can explore other options if necessary.

Reply:
+1 to this

we should be aiming for defined interfaces, but there are cases like this where generic types are perfect for (especially to avoid an additional casting step later on). we need to find a balance of interfaces and generic types that reduce risks, rather than stray too far to one side